src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,44 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+export const TodoModal: React.FC<{
+  initialTodo: Todo
+  setCarierTodo: React.Dispatch<React.SetStateAction<Todo>>
+  carierTodo: Todo
+}> = ({
+  initialTodo,
+  setCarierTodo,
+  carierTodo,
+}) => {
+  const [user, setUser] = useState<User>();
+
+  const {
+    title, userId, id, completed,
+  } = carierTodo;
+
+  useEffect(() => {
+    try {
+      getUser(userId)
+        .then((userFromServer) => {
+          setUser(userFromServer);
+        });
+    } catch (error) {
+      <Loader />;
+    }
+  }, [userId]);
+
+  const resetTodoModal = () => {
+    setCarierTodo(initialTodo);
+  };
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!user ? (
```

Comment:
it's better to check the `carierTodo`
```suggestion
      {!carierTodo ? (
```

Reply:
Use this method, I check that user must be not undefined.

Reply:
I cant. User must see Loader when information from user dont come. I dont wait carierTodo, only user.